source/contracts/libraries/collections/Map.sol

```diff
@@ -0,0 +1,53 @@
+pragma solidity 0.4.17;
+
+import 'libraries/DelegationTarget.sol';
+import 'libraries/Ownable.sol';
+import 'libraries/Initializable.sol';
+
+
+contract Map is DelegationTarget ,Ownable, Initializable {
+    mapping(bytes32 => address) private items;
```

Comment:
Consider making this `bytes32 => bytes32` and providing function overloads for the different 256-bit types (insertion) and getters that do casting for the user:
```
function add(bytes32 _key, bytes32 _value) ... { items[_key] = _value; ... }
function add(bytes32 _key, address _value) ... { return add(_key, bytes32(_value)); }
function add(bytes32 _key, uint256 _value) ... { return add(_key, bytes32(_value)); }
function add(uint256 _key, address _value) ... { return add(bytes32(_key), bytes32(_value)); }
function getAsAddress(bytes32 _key) ... { return address(items[_key]); }
function getAsAddress(uint256 _key) ... { return address(items[bytes32(_key])); }
```

I recommend initially only providing `bytes32, bytes32` overloads and then adding additional overloads as needed (it looks like in this case you need `bytes32 => address`) since there are so many possible combinations.  This allows us to keep `Map` as a generic/general purpose construct, not tightly coupled to its initial usage (in Market).

Reply:
Will do